Backports for 1.2.0 release

KristofferC opened the pull request:
Backported PRs:
- [x] #32007 - Sparse matrix: fix fast implementation of findnext and findprev for cartesian coordinates
- [x] #32476 - fix #32467, parsing macro on `for` inside `do` inside call
- [x] #32509 - fix #32488, type intersection regression
- [x] #32605 - Fix #32579 - Issue in typeconstraint accumulation
- [x] #32623 - fix #32620, crash when assigning to static parameter
- [x] #31698 - Fix some vararg-related subtyping issues
- [x] #32590 - fix #32582, type intersection bug in unions
- [x] #32321 - codegen,gc-lowering: post fixup tbaa information
- [x] #32271 - [Statistics] fix type determination in corm
- [x] #32145 - fix bug in block renaming for dead code elimination
- [x] #32731 - fix #32703, bad ambiguity error 

Need manual backport:
- [x] #32038 - Update Julia logos, removing rings, round 2
- [x] #32054 - Update Windows icon and NSIS banner

Non-merged PRs with backport label:
- [ ] #32327 - bug fixes in matrix log

musm committed:
Remove unused download of 7z-extra

(cherry picked from commit 667740697b82b0dc62a1b081f821f1fea643565b)

musm committed:
Improve some aspects of the NSIS installer (#31959)

(cherry picked from commit 9d6fffee937b3cbf44bcf53da1b69f1c7848c96f)

cormullion committed:
update Julia logos, removing rings round 2 (#32038)

(cherry picked from commit b5a4601bbe8a6f1b8ce8f4e307543850c08adb8e)

Conflicts:
	contrib/mac/frameworkapp/installresources/logo_hires.png

Keno committed:
Fix some vararg-related subtyping issues (#31698)

* Fix jl_obvious_subtype with INT vararg constraint

* Fix a vararg-related non-transitivity in subtyping

* Fix another vararg subtype issue

* Take advantage of their being at most one UnionAll wrapped around a Vararg

Upon construction, we normalize `Vararg{T, N} where {T,N}` to `Vararg{T where T, N} where N`,
thus there can be at most one UnionAll wrapper around a Vararg. In subtyping we were already
assuming that there can be at most two such wrappers, so simply adjust that and add an
appropriate assertion to make sure we catch any cases where this ever goes wrong.

* Rewrite subtype_tuple to fix extra cases

* Put back the case for naked varargs

* Update test/subtype.jl

Co-Authored-By: Keno <keno@alumni.harvard.edu>

* Add test for #31805

* Fix style review comments

* Rename variable

* In person review changes

* Fix bug

* Handle integer bounds on left arguments in the environment

In subtyping proper, variables introduced on the left (i.e. forall
variables) don't have any equality constraints, because we have no
syntax for creating them. However, intersection does sometimes create
such environments, so we need to handle it in subtyping.

(Cherry-picked from 4a38e790852318540d2e89ad5864c7a3633a7909)

musm committed:
Update Windows icon and NSIS banner (#32054)

(cherry picked from commit a43c46b68796228838041a140356f96e6f76184b)

tkluck committed:
Sparse matrix: fix fast implementation of findnext and findprev for cartesian coordinates (#32007)

Revert "sparse findnext findprev hash performance improved (#31354)"

This seems to duplicate work from #23317 and it causes performance
degradation in the cases that one was designed for. See
https://github.com/JuliaLang/julia/pull/31354#issuecomment-491511974

This reverts commit e0bef65d5ac228a0e84e0cee4013c245ce2009eb.

Thanks to @mbauman for spotting this issue in
https://github.com/JuliaLang/julia/pull/32007#issuecomment-492018400.

(cherry picked from commit ec797ef984fc1f56a94c851ab720eb5505e85119)

musm committed:
Minor changes to windows installer, icons, and shortcuts

(cherry picked from commit 6998c4ac5feb6ef06ef270d69d5706a3b14cf0e7)

yhls committed:
fix bug in block renaming for dead code elimination

(cherry picked from commit 3829e8fd475e4b468865cbd21133c5fec34951df)

dkarrasch committed:
bugfix for ldiv!(D::Diagonal, B::StridedVecOrMat) and tests (#32104)

(cherry picked from commit 85603e15a025b011a95e705d6a43117cb478e30a)

vtjnash committed:
codegen,gc-lowering: post fixup tbaa information (#32321)

Certain metadata (invariant.load, tbaa_const) allows reordering of
memory operations over unknown calls (including safepoint). Before GC
lowering this is legal, because the optimizer sinking a load over a
safepoint will automatically extend the live range of the heap object
(by definition the live range of the heap object before GC lowering
matches the live ranges of any reference to it). However, after GC
lowering, we have fixed the live ranges of the heap objects, and thus it
is no longer legal to sink such memory operations over safepoints (as
the update to the object's live range won't be tracked).

fix #32215

(cherry picked from commit b2304c528a2b055bfe6e88a62b773765889259e9)

JeffBezanson committed:
fix #32467, parsing macro on `for` inside `do` inside call (#32476)

(cherry picked from commit b74ef058a39866852707f877cfc78c1a5cf00247)

JeffBezanson committed:
fix #32488, type intersection regression (#32509)

Issue was caused by 74305bf2c43546bf76843b72ff0d268a3fe16920, so this
reverts part of the new logic from there.
(cherry picked from commit a9ad6c2bb5771e54a3d4505e2d3146d280c3b093)

musm commented:
Might as well backport https://github.com/JuliaLang/julia/pull/32054  if the branding also got backported ?

ViralBShah commented:
Yes that would be a good idea.

Keno committed:
Fix #32579 - Issue in typeconstraint accumulation (#32605)

The bug here is a bit subtle, but perhaps best illustrated with
the included test case:
```
function f32579(x::Int64, b::Bool)
    if b
        x = nothing
    end
    if isa(x, Int64)
        y = x
    else
        y = x
    end
    if isa(y, Nothing)
        z = y
    else
        z = y
    end
    return z === nothing
end
```
The code just after SSA conversion looks like:
```
2  1 ─       goto #3 if not _3
3  2 ─ %2  = Main.nothing::Core.Compiler.Const(nothing, false)
5  3 ┄ %3  = φ (#2 => %2, #1 => _2)::Union{Nothing, Int64}
   │   %4  = (%3 isa Main.Int64)::Bool
   └──       goto #5 if not %4
6  4 ─ %6  = π (%3, Int64)
   └──       goto #6
8  5 ─ %8  = π (%3, Nothing)
10 6 ┄ %9  = φ (#4 => %6, #5 => %8)::Union{Nothing, Int64}
   │   %10 = (%9 isa Main.Nothing)::Bool
   └──       goto #8 if not %10
11 7 ─ %12 = π (%9, Nothing)
   └──       goto #9
13 8 ─ %14 = π (%9, Int64)
15 9 ┄ %15 = φ (#7 => %12, #8 => %14)::Union{Nothing, Int64}
   │   %16 = (%15 === Main.nothing)::Bool
   └──       return %16
```
Now, we have special code in SROA (despite it not really being an
SROA transform) that looks at `===` and replaces
it by a nest of phis of booleans. The reasoning for this transform
is that it eliminates a use of a value where we only care about the
type and not the content, thus making it more likely that the value
will subsequently be eligible for SROA. In addition, while it goes
along resolving which values feed into any particular phi, it
accumulates and type conditions it encounters along the way.

Thus in the example above, something like the following happens:
- We look at %14, which πs to %9 with an Int64 constraint, so we only
  consider the #4 predecessor for %9 (due to the constraint), until
  we get to %3, where we again only consider the #1 predecessor,
  where we find the argument (of type Int64) and conclude the result
  is always false
- Now we pop the next item of the stack from our original phi, look
  at %12, which πs to %9 with a Nothing constraint.

At this point we used to terminate the search because we already looked
at %9. However, crucially, we looked at %9 only with an Int64 constraint,
so we missed the fact that `nothing` was in fact a possible value for this
phi. The result was a missing entry in the generated phi node:
```
1 ─       goto #3 if not b
2 ─ %2  = Main.nothing::Core.Compiler.Const(nothing, false)
3 ┄ %3  = φ (#1 => false)::Bool
│   %4  = φ (#2 => %2, #1 => _2)::Union{Nothing, Int64}
│   %5  = (%4 isa Main.Int64)::Bool
└──       goto #5 if not %5
4 ─ %7  = π (%4, Int64)
└──       goto #6
5 ─ %9  = π (%4, Nothing)
6 ┄ %10 = φ (#4 => %3, #5 => %3)::Bool
│   %11 = φ (#4 => %7, #5 => %9)::Union{Nothing, Int64}
│   %12 = (%11 isa Main.Nothing)::Bool
└──       goto #8 if not %12
7 ─       goto #9
8 ─       nothing::Nothing
9 ┄ %16 = φ (#7 => %10, #8 => %10)::Bool
└──       return %16
```
(note the missing #2 predecessor in phi node %3), which would result
in an undefined value at runtime, though in this case LLVM would
have taken advantage of that to just return 0:
```
define i8 @julia_f32579_16051(i64, i8) {
top:
;  @ REPL[1]:15 within `f32579'
  ret i8 0
}
```
Compare this now to the optimized IR with this patch:
```
1 ─       goto #3 if not b
2 ─ %2  = Main.nothing::Core.Compiler.Const(nothing, false)
3 ┄ %3  = φ (#2 => true, #1 => false)::Bool
│   %4  = φ (#2 => %2, #1 => _2)::Union{Nothing, Int64}
│   %5  = (%4 isa Main.Int64)::Bool
└──       goto #5 if not %5
4 ─ %7  = π (%4, Int64)
└──       goto #6
5 ─ %9  = π (%4, Nothing)
6 ┄ %10 = φ (#4 => %3, #5 => %3)::Bool
│   %11 = φ (#4 => %7, #5 => %9)::Union{Nothing, Int64}
│   %12 = (%11 isa Main.Nothing)::Bool
└──       goto #8 if not %12
7 ─       goto #9
8 ─       nothing::Nothing
9 ┄ %16 = φ (#7 => %10, #8 => %10)::Bool
└──       return %16
```
The %3 phi node has its missing entry and the generated LLVM code
correctly returns `b`:
```
define i8 @julia_f32579_16112(i64, i8) {
top:
  %2 = and i8 %1, 1
;  @ REPL[1]:15 within `f32579'
  ret i8 %2
}
```

(cherry picked from commit 0a12944ccaf175638ff05085e0519230fd05f157)

JeffBezanson committed:
fix #32620, crash when assigning to static parameter (#32623)

(cherry picked from commit b9b099f278e2f344821ba63998f6d5c75fe66827)

JeffBezanson committed:
fix #32582, type intersection bug in unions (#32590)

Caused by 4f8a7b965a98c307f48820c35ca98f9d754096a9; reverts part
of that.
(cherry picked from commit ddd3f0fb022b812978467aa430346f32f87ce7f7)

ararslan commented:
@nanosoldier `runbenchmarks(ALL, vs=":release-1.2")`

nanosoldier commented:
[Your benchmark job](https://github.com/JuliaLang/julia/pull/32592#issuecomment-513894603) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/BaseBenchmarkReports/blob/ab70ec7bf287257a801846331ea6c04ac6b79afe/71f414f_vs_9248bf7/report.md). cc @ararslan

ararslan commented:
PkgEval failures:

- [ ] [AlgebraicMultigrid](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-algebraicmultigrid-log)
- [ ] [BSON](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-bson-log)
- [ ] [BlackBoxOptim](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-blackboxoptim-log)
- [ ] [Cbc](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-cbc-log)
- [ ] [Cumulants](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-cumulants-log)
- [ ] [DataFrames](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-dataframes-log)
- [ ] [Discretizers](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-discretizers-log)
- [ ] [DistQuads](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-distquads-log)
- [ ] [DistributedArrays](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-distributedarrays-log)
- [ ] [DutyCycles](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-dutycycles-log)
- [ ] [GLMNet](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-glmnet-log)
- [ ] [HypothesisTests](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-hypothesistests-log)
- [ ] [InfoZIP](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-infozip-log)
- [ ] [Jacobi](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-jacobi-log)
- [ ] [LanguageServer](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-languageserver-log)
- [ ] [Libtask](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-libtask-log)
- [ ] [MLDataPattern](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-mldatapattern-log)
- [ ] [MatrixNetworks](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-matrixnetworks-log)
- [ ] [Millboard](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-millboard-log)
- [ ] [Mixers](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-mixers-log)
- [ ] [NamedArrays](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-namedarrays-log)
- [ ] [NearestNeighborDescent](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-nearestneighbordescent-log)
- [ ] [Nullables](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-nullables-log)
- [ ] [OSQP](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-osqp-log)
- [ ] [PkgTemplates](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-pkgtemplates-log)
- [ ] [PolynomialRings](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-polynomialrings-log)
- [ ] [PriorityChannels](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-prioritychannels-log)
- [ ] [ResumableFunctions](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-resumablefunctions-log)
- [ ] [Revise](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-revise-log)
- [ ] [RingBuffers](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-ringbuffers-log)
- [ ] [Sched](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-sched-log)
- [ ] [StaticArrays](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-staticarrays-log)
- [ ] [StrBase](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-strbase-log)
- [ ] [Strided](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-strided-log)
- [ ] [Traceur](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-traceur-log)
- [ ] [UnicodePlots](https://gist.github.com/ararslan/0349ea845a62d892e0242113074035ed#file-unicodeplots-log)

quinnj commented:
The DataFrames.jl failure looks like https://github.com/JuliaLang/julia/issues/32607

ararslan commented:
The Cbc issue was addressed in https://github.com/JuliaOpt/Cbc.jl/pull/115 and made it into a tag but it's still failing with the same error.

KristofferC commented:
Regarding Cbc, they directly include a test from another package which happens to use Compat: https://github.com/JuliaOpt/Cbc.jl/blob/8fe080bdc21265c7b4b14c39c70c8848823927a8/test/MPB_wrapper.jl#L5 so they do need Compat...

ararslan commented:
Fix for Cumulants: https://github.com/ZKSI/Cumulants.jl/pull/14

ararslan commented:
Fix for Cbc: https://github.com/JuliaOpt/Cbc.jl/pull/118

ararslan commented:
Looks like BlackBoxOptim is just a printing change

ararslan commented:
MLDataPattern is the same as previous runs: https://github.com/JuliaLang/julia/pull/31727#issuecomment-493047568 cc @Evizero

ararslan commented:
UnicodePlots issue persists: https://github.com/JuliaLang/julia/pull/31727#issuecomment-492382769

ararslan commented:
NearestNeighborDescent also seems like an inference regression.

dkarrasch committed:
[Statistics] fix type determination in corm (#32271)

* [Statistics] fix type determination in corm

* remove obsolete typeof

* use first element(s) for type initialization

* add test for inhomogeneous data and for overflow

* fix test with NaN

(cherry picked from commit a8a567e72b669525dc4d3a38af275fe042f44a42)

alyst commented:
FYI: robertfeldt/BlackBoxOptim.jl#128 should fix the PkgEval failure

ararslan commented:
PolynomialRings and StrBase are new method ambiguities.

ararslan commented:
So supposedly the DataFrames issue was fixed by https://github.com/JuliaLang/julia/pull/31698, but that change was backported and the issue persists. In chatting with @vtjnash, it's possible that #31698 fixes it but depends on other things.

ararslan commented:
LanguageServer failure looks real. Also broken on nightly.

ararslan commented:
PolynomialRings master fails its own tests on 1.1 and later.

rofinn commented:
I'd say that you can probably skip/ignore PkgTemplates.jl. Those tests need a lot of work and we can't switch off 1.0 right away anyways.

dkarrasch committed:
fix generic ldiv! for CholeskyPivoted (#32593)

* fix generic ldiv! for CholeskyPivoted

* add back one empty line to make backport easier

* don't interfere with previous tests

(cherry picked from commit d0e11cf028c7d62a87b2906f5662283df87c7600)

garrison committed:
Implement isdiag(::Bidiagonal), add tests (#32694)

(cherry picked from commit 25eb8c6940b0ec664257567e166c316107b04e23)

JeffBezanson committed:
mention sprand change in NEWS

backports abae4a8a912de7e901f410755b329c38bf7582d1

JeffBezanson commented:
NearestNeighborDescent is due to the following change in type intersection:
```
julia> typeintersect(Tuple{Vector{Vector{Float32}},Matrix,Matrix}, Tuple{Vector{V},Matrix{Int},Matrix{S}} where {S, V<:AbstractVector{S}})
Tuple{Array{Array{Float32,1},1},Array{Int64,2},Array{S,2}} where S
```
Used to give a concrete type. Same cause as #32703 (5283847), but not as easy to fix. However weakening type information is allowed, and was done in this case to fix crashes, so we might not fix this for 1.2.

JeffBezanson commented:
Backported #32321, but I was not able to get its test working since there have also been many other changes to gc frame and allocation lowering.

dkarrasch committed:
fix test for sparse matrix assignment (#32756)

(cherry picked from commit 30b36364b03a8afe3c543c0a040f3611db7ae19c)

JeffBezanson committed:
fix #32703, bad ambiguity error (#32731)

caused by 52838478aabe93120a0152ec5da733c9d8f0c39a
(cherry picked from commit dc0bf52d0bd8ccfab414d1f465569fdd0755ae73)

JeffBezanson committed:
codegen: ensure required attributes get set on all functions (#32772)

It is especially important that we set the stack-alignment attribute on Win32 on cfunction entry.
(cherry picked from commit d8b5118c4c7c95d5f268b5696b37032443a43c12)

JeffBezanson committed:
fix #32607, bug in typeintersect of `X{<:Tuple1}` vs. `X{Tuple2}` (#32771)

(cherry picked from commit 4671ddc54f1711f80349721e705b66c17f60f37e)

JeffBezanson committed:
fix #32777, regression in type intersection of unions (#32788)

Fortunately, #32771 allows reverting the offending change.
(cherry picked from commit c158ff24e5dd94f6f5b6798f670a4a4995c731ad)

JeffBezanson commented:
Requesting help backporting #32054. @musm ?

musm commented:
Do perform a backport is the procedure to cherry pick onto  the `release-1.2` branch?

JeffBezanson commented:
Cherry pick with `-x` onto this branch, `backports-1.2.0`.

ararslan commented:
```bash
git pull
git checkout -b backport-32054 origin/backports-1.2.0
git cherry-pick -x e a43c46b68796228838041a140356f96e6f76184b
# Fix conflicts
git push musm backport-32054
# Open PR against this branch
```

ararslan committed:
Merge pull request #32819 from musm/backport-32054

Backport 32054

ararslan commented:
New PkgEval run, failures (most are probably the same as above):

- [ ] [AlgebraicMultigrid](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-algebraicmultigrid-log)
- [ ] [ArnoldiMethod](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-arnoldimethod-log)
- [ ] [BSON](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-bson-log)
- [ ] [BitBasis](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-bitbasis-log)
- [ ] [Cumulants](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-cumulants-log)
- [ ] [DASSL](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-dassl-log)
- [ ] [DBFTables](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-dbftables-log)
- [ ] [Discretizers](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-discretizers-log)
- [ ] [DistQuads](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-distquads-log)
- [ ] [DistributedArrays](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-distributedarrays-log)
- [ ] [Dolang](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-dolang-log)
- [ ] [DutyCycles](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-dutycycles-log)
- [ ] [Flux](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-flux-log)
- [ ] [GPUifyLoops](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-gpuifyloops-log)
- [ ] [GenericLinearAlgebra](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-genericlinearalgebra-log)
- [ ] [HypothesisTests](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-hypothesistests-log)
- [ ] [InfoZIP](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-infozip-log)
- [ ] [IntervalArithmetic](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-intervalarithmetic-log)
- [ ] [Jacobi](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-jacobi-log)
- [ ] [LanguageServer](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-languageserver-log)
- [ ] [LocalFunctionApproximation](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-localfunctionapproximation-log)
- [ ] [MLDataPattern](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-mldatapattern-log)
- [ ] [MatrixNetworks](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-matrixnetworks-log)
- [ ] [Millboard](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-millboard-log)
- [ ] [Mixers](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-mixers-log)
- [ ] [ModelingToolkit](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-modelingtoolkit-log)
- [ ] [NLSolversBase](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-nlsolversbase-log)
- [ ] [NLopt](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-nlopt-log)
- [ ] [NamedArrays](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-namedarrays-log)
- [ ] [NearestNeighborDescent](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-nearestneighbordescent-log)
- [ ] [Nullables](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-nullables-log)
- [ ] [OSQP](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-osqp-log)
- [ ] [ParameterizedFunctions](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-parameterizedfunctions-log)
- [ ] [PkgTemplates](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-pkgtemplates-log)
- [ ] [PriorityChannels](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-prioritychannels-log)
- [ ] [QSWalk](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-qswalk-log)
- [ ] [QuantumLattices](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-quantumlattices-log)
- [ ] [ResumableFunctions](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-resumablefunctions-log)
- [ ] [Revise](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-revise-log)
- [ ] [RingBuffers](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-ringbuffers-log)
- [ ] [Rotations](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-rotations-log)
- [ ] [SQLite](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-sqlite-log)
- [ ] [Sched](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-sched-log)
- [ ] [SolverTools](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-solvertools-log)
- [ ] [Strided](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-strided-log)
- [ ] [TensorCast](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-tensorcast-log)
- [ ] [Traceur](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-traceur-log)
- [ ] [UnicodePlots](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-unicodeplots-log)
- [ ] [WaterFlows](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-waterflows-log)

1168 packages were skipped on 1.1, 1214 were skipped on this branch.

vtjnash commented:
It seems like the possibly interesting ones there to look into more closely are:
[Dolang](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-dolang-log)
[SolverTools](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-solvertools-log)
[SQLite](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-sqlite-log)
[LanguageServer](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-languageserver-log)
[MatrixNetworks](https://gist.github.com/ararslan/6a4c76e4319001be4e0f7a79f3407aec#file-matrixnetworks-log)

quinnj commented:
I don't think SQLite is actually a problem; it uses serialization for custom types, so if anything gets tweaked there, it breaks.

ararslan commented:
I've edited my comment above with the results of a run which does not skip packages for which a dependency fails tests. I went through and checked off the usual suspects from earlier and submitted a bunch of PRs to fix the new minor things. There are a couple of failures I don't understand in addition to the ones Jameson noted above.

JeffBezanson commented:
Rotations: just some changed printing

JeffBezanson commented:
ArnoldiMethod: is getting different numbers. Needs to be looked at by somebody who knows about numbers.

JeffBezanson commented:
QuantumLattices: some block expression now includes a location that didn't previously

JeffBezanson commented:
RandomMatrices: also different numbers

JeffBezanson commented:
GPUifyLoops: Needs a test dependency on InteractiveUtils?

JeffBezanson commented:
IntervalArithmetic: Inference widened to Any

JeffBezanson commented:
Dolang and SolverTools both reduced to this intersection regression:
```
julia> typeintersect(Tuple{Ref{DataType},Any}, Tuple{Ref{B},Int} where B<:Type)
Union{}
```

vtjnash commented:
IntervalArithmetic is a known package issue (https://github.com/JuliaIntervals/IntervalArithmetic.jl/issues/291)

JeffBezanson committed:
fix type intersection bug affecting Dolang.jl and SolverTools.jl (#32853)

(cherry picked from commit 5dbcf20e64bd91474aa24840f63c63f2c9e2a712)

vchuravy commented:
https://github.com/JuliaLang/julia/pull/32592#issuecomment-520034703 fixed GPUifyLoops `Pkg.test`.
x-ref https://github.com/JuliaRegistries/General/pull/2626

ararslan commented:
According to @andreasnoack, ArnoldiMethod and RandomMatrices may be because we now sort eigenvalues.

andreasnoack commented:
ArnoldiMethod failures are fixed by https://github.com/haampie/ArnoldiMethod.jl/pull/99

JeffBezanson commented:
Ok, great. In that case I think we're ready to merge this and put out the next RC, which I have high hopes will be final.

ararslan commented:
We should backport https://github.com/JuliaLang/julia/pull/32853 first. Also note that LanguageServer seems like it could be a real regression.

JeffBezanson commented:
I think LanguageServer is fixed: https://github.com/julia-vscode/LanguageServer.jl/issues/372

ararslan commented:
Excellent! I looked at MatrixNetworks but couldn't figure out what the deal was, and I think that's the last unchecked item.

andreasnoack commented:
Master of `MatrixNetworks` passes for me on 1.3 so I think they only need to make a new release. The latest one is from October.

ararslan commented:
You're right, MatrixNetworks master passes locally for me on this branch. Thanks!

KristofferC commented:
We put quite a lot of commits here without a nanosoldier run, so running one here for posterity.

@nanosoldier `runbenchmarks(ALL, vs="@9248bf7687a91fa0cce4bd496889e5cf578b009d")`

nanosoldier commented:
[Your benchmark job](https://github.com/JuliaLang/julia/pull/32592#issuecomment-522084244) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/BaseBenchmarkReports/blob/ebd87055e7cadbcda92395215fd18dc2819565f4/bd375ac_vs_9248bf7/report.md). cc @ararslan
